### PR TITLE
Some DRM providers require default KID in wrapped license requests

### PR DESCRIPTION
**Pull request by birme**
Suggested fix for #528

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -31,3 +31,4 @@ SameGoal Inc. <*@samegoal.com>
 Sanborn Hilland <sanbornh@rogers.com>
 TalkTalk Plc <*@talktalkplc.com>
 uStudio Inc. <*@ustudio.com>
+Jonas Birmé <jonas.birme@eyevinn.se>
```
Comment (joeyparrish) [inline AUTHORS]: Please keep this list alphabetized.

File: CONTRIBUTORS [hunk 2]
```diff
@@ -48,3 +48,4 @@ Thomas Stephens <thomas@ustudio.com>
 Timothy Drews <tdrews@google.com>
 Vasanth Polipelli <vasanthap@google.com>
 Vignesh Venkatasubramanian <vigneshv@google.com>
+Jonas Birmé <jonas.birme@eyevinn.se>
```
Comment (joeyparrish) [inline CONTRIBUTORS]: Please keep this list alphabetized.

File: lib/media/drm_engine.js [hunk 3]
```diff
@@ -764,6 +778,10 @@ shaka.media.DrmEngine.prototype.processDrmInfos_ =
         }
       });
     }
+
+    if (drmInfo.defaultKeyID) {
```
Comment (joeyparrish) [inline lib/media/drm_engine.js]: This is inconsistent with other places where you called it defaultKID, but as mentioned earlier, please rename to keyIds.

File: externs/shaka/manifest.js [hunk 4]
```diff
@@ -136,7 +136,8 @@ shakaExtern.InitDataOverride;
  *   audioRobustness: string,
  *   videoRobustness: string,
  *   serverCertificate: Uint8Array,
- *   initData: Array.<!shakaExtern.InitDataOverride>
+ *   initData: Array.<!shakaExtern.InitDataOverride>,
+ *   defaultKID: ?string
```
Comment (joeyparrish) [inline externs/shaka/manifest.js]: There can be (and best practice for content authoring dictates there should be) multiple key IDs in a single manifest.  If you're going to go this route, this should be an array of strings (default empty list, not null), and it should be called keyIds.

**Comment by birme** [comment]
Thank you for the feedback. I will correct these things

File: lib/media/drm_engine.js [hunk 5]
```diff
@@ -723,10 +740,11 @@ shaka.media.DrmEngine.prototype.createCurrentDrmInfo_ = function(
  * @param {!Array.<string>} licenseServers
  * @param {!Array.<!Uint8Array>} serverCerts
  * @param {!Array.<!shakaExtern.InitDataOverride>} initDatas
+ * @param {!Array.<string>} defaultKeys
```
Comment (joeyparrish) [inline lib/media/drm_engine.js]: For consistency, let's give this the same name as the structure member: keyIds.

File: lib/media/drm_engine.js [hunk 6]
```diff
@@ -685,7 +697,11 @@ shaka.media.DrmEngine.prototype.createCurrentDrmInfo_ = function(
   /** @type {!Array.<!shakaExtern.InitDataOverride>} */
   var initDatas = [];
 
-  this.processDrmInfos_(drmInfos, licenseServers, serverCerts, initDatas);
+  /** @type {!Array.<!string>} */
```
Comment (joeyparrish) [inline lib/media/drm_engine.js]: strings are non-nullable by default, so you can just say `!Array.<string>`

**Comment by birme** [comment]
No probs, fixed that last thing now

**Comment by joeyparrish** [comment]
Looks like you'll need to update the tests to match.  You can run the tests for yourself with `python build/test.py`.

**Comment by birme** [comment]
Chrome 53.0.2785 (Mac OS X 10.10.5): Executed 891 of 927 (skipped 36) SUCCESS (6 mins 11.643 secs / 7 mins 27.425 secs)

**Comment by birme** [comment]
Sorry, thought I did a lint check before pushing. Will fix

**Pull request merged**